src/systems/menu_impl.cpp

```diff
@@ -225,148 +349,303 @@ static void draw_selectable_highlight(float y) {
+                             "Bind");
+            if (widget.input_select.can_unbind) {
+                draw::debug_text(START + 1 * BLOCK_WIDTH, Y_HEIGHT, draw::GRAY, "Y");
+                draw::debug_text(BUTTON_START + 2 * BLOCK_WIDTH, Y_HEIGHT, draw::WHITE, ":");
+                draw::debug_text(BUTTON_START + 2 * BLOCK_WIDTH + HALF_SPACE, Y_HEIGHT, draw::WHITE,
+                                 "Unbind");
+            }
+            draw::debug_text(START + 2 * BLOCK_WIDTH, Y_HEIGHT, draw::GRAY, "X");
+            draw::debug_text(BUTTON_START + 3 * BLOCK_WIDTH, Y_HEIGHT, draw::WHITE, ":");
+            draw::debug_text(BUTTON_START + 3 * BLOCK_WIDTH + HALF_SPACE, Y_HEIGHT, draw::WHITE,
+                             "Reset");
+            break;
+        }
+        default: {
+            break;
+        }
+    }
+}
 
-    for (u32 i = 0; i < menu->num_widgets; i++) {
-        Widget& widget = menu->widgets[i];
+void draw_widget(Widget& widget, u32 selected_idx, u32* selectable_idx, u32* y,
+                 mkb::GXColor lerped_color) {
+    switch (widget.type) {
+        case WidgetType::HideableGroupWidget: {
+            if (show_hideable_widget(&widget)) {
+                for (u32 i = 0; i < widget.hideable_group.num_widgets; i++) {
+                    Widget& w = widget.hideable_group.widgets[i];
+                    draw_widget(w, selected_idx, selectable_idx, y, lerped_color);
+                }
+            }
+            break;
+        }
+        case WidgetType::Header: {
+            draw::debug_text(MARGIN + PAD, *y, draw::ORANGE, widget.header.label);
+            *y += LINE_HEIGHT;
+            break;
+        }
+        case WidgetType::Text: {
+            draw::debug_text(MARGIN + PAD, *y, draw::WHITE, widget.text.label);
+            *y += LINE_HEIGHT;
+            break;
+        }
+        case WidgetType::ColoredText: {
+            draw::debug_text(MARGIN + PAD, *y, widget.colored_text.color,
+                             widget.colored_text.label);
+            *y += LINE_HEIGHT;
+            break;
+        }
+        case WidgetType::Checkbox:
+        case WidgetType::GetSetCheckbox: {
+            const char* label = nullptr;
+            bool value = false;
+            if (widget.type == WidgetType::Checkbox) {
+                label = widget.checkbox.label;
+                value = pref::get(widget.checkbox.pref);
+            } else {
+                label = widget.get_set_checkbox.label;
+                value = widget.get_set_checkbox.get();
+            }
 
-        switch (widget.type) {
-            case WidgetType::Header: {
-                draw::debug_text(MARGIN + PAD, y, draw::ORANGE, widget.header.label);
-                y += LINE_HEIGHT;
-                break;
+            if (selected_idx == *selectable_idx) {
+                draw_selectable_highlight(*y);
+                draw_help(widget);
```

Comment:
This is called in a lot of places. Can we just call `draw_help_layout()` / `draw_help()` a single time in `draw()` and pass in the currently selected widget? It's probably not too slow to call `get_selected_widget()` another time.

Reply:
Will move to disp